src/Bridge/Flysystem/FilesystemWrapper.php

```diff
@@ -0,0 +1,136 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Externals\Bridge\Flysystem;
+
+use EoneoPay\Externals\Filesystem\Exceptions\FileNotFoundException;
+use EoneoPay\Externals\Filesystem\Interfaces\FilesystemInterface;
+use League\Flysystem\FileNotFoundException as FlysystemFileNotFoundException;
+use League\Flysystem\FilesystemInterface as FlysystemInterface;
+
+class FilesystemWrapper implements FilesystemInterface
+{
+    /**
+     * @var \League\Flysystem\FilesystemInterface
+     */
+    private $flysystem;
+
+    /**
+     * FilesystemWrapper constructor.
+     *
+     * @param \League\Flysystem\FilesystemInterface $flysystem
+     */
+    public function __construct(FlysystemInterface $flysystem)
+    {
+        $this->flysystem = $flysystem;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \League\Flysystem\FileNotFoundException
+     */
+    public function append(string $filename, string $contents): bool
+    {
+        /** @var resource $existing */
+        $existing = $this->readStream($filename);
+
+        /** @var resource $writeStream */
+        $writeStream = \fopen('php://memory', 'ab+');
+
+        \stream_copy_to_stream($existing, $writeStream);
+        \fwrite($writeStream, $contents);
+
+        return $this->flysystem->updateStream($filename, $writeStream);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function exists(string $filename): bool
+    {
+        return $this->flysystem->has($filename);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function files(?string $directory = null, ?bool $recursive = null): array
+    {
+        return $this->flysystem->listContents($directory ?? '', $recursive ?? false);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function path(?string $filename = null): string
+    {
+        return '/' . $filename;
```

Comment:
❓

Reply:
It made sense the time - but I'm not entirely sure it's resolvable. Flysystem does not expose any kind of prefix about the underlying adapters.

Reply:
Discussion from slack is to remove this from the interface - but, this will break the current Loader tests which use the `vfs://` filesystem for testing.
ie; `EoneoPay\Externals\Environment\Loader` relies on the current path to the `.env` file being passed in as a string.

The right way to solve is probably to rewrite the loader to accept a `EoneoPay\Externals\Filesystem\Interfaces\FilesystemInterface`.

Reply:
Created https://eonx.atlassian.net/browse/PYMT-1581 & https://eonx.atlassian.net/browse/PYMT-1582